server/src/main/java/org/opensearch/indices/replication/common/ReplicationLuceneIndex.java

```diff
@@ -361,6 +370,10 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
         }
 
         public void addFileDetails(String name, long length, boolean reused) {
+            // if set of files contains a segments file, it represents an incoming commit point
+            if (name.startsWith(IndexFileNames.SEGMENTS)) {
+                this.hasIncomingCommitPoint = true;
```

Comment:
Should this be set to false in the `clear()` method? Alternatively, this could just be computed scanning the list of files when requested. It's obviously less efficient, but you avoid potential bugs and is simpler.

Reply:
+1.  I think another alternative to scanning when requested if we are worried about efficiency is scanning the list of files in `SegmentReplicationTarget` [here](https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/indices/replication/SegmentReplicationTarget.java#L181) after all files have been added and set a flag on the ReplicationLuceneIndex that is segrep secific?

Reply:
@Poojita-Raj Another option here if this is segrep specific - we can disable the fsync entirely in MultiFileWriter with segrep enabled, then in `SegmentReplicationTarget` we can fetch all files, loop through them, and fsync if needed?